source/common/router/config_impl.cc

```diff
@@ -269,9 +269,14 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
       priority_(ConfigUtility::parsePriority(route.route().priority())),
       total_cluster_weight_(
           PROTOBUF_GET_WRAPPED_OR_DEFAULT(route.route().weighted_clusters(), total_weight, 100UL)),
+      route_action_request_headers_parser_(
+          HeaderParser::configure(route.route().request_headers_to_add())),
+      route_action_response_headers_parser_(
+          HeaderParser::configure(route.route().response_headers_to_add(),
+                                  route.route().response_headers_to_remove())),
       request_headers_parser_(HeaderParser::configure(route.route().request_headers_to_add())),
```

Comment:
It seems we try to configure parsers with `route.route().request_headers_to_add()` twice here, hence the relevant integration tests are failed. 

Do you want to configure `request_headers_parser_` by `route.request_headers_to_add()` instead? And regarding that, I think the newly added fields to `Route` (`request_headers_to_add`, `response_headers_to_add`, and `response_headers_to_remove`) needs to be accomodated in the test.